internal/v2/security.go

```diff
@@ -57,16 +57,15 @@ func (s *SecurityProviderExoscaleV2) signRequest(req *http.Request, expiration t
 	sigParts = append(sigParts, fmt.Sprintf("%s %s", req.Method, req.URL.Path))
 	headerParts = append(headerParts, "EXO2-HMAC-SHA256 credential="+s.apiKey)
 
-	// Request body
-	body := ""
+	// Request body if present
 	if req.Body != nil {
 		data, err := ioutil.ReadAll(req.Body)
 		if err != nil {
 			return err
 		}
-		body = string(data)
+		sigParts = append(sigParts, string(data))
+		req.Body = newCopiedBody(data)
```

Comment:
```suggestion
		req.Body = ioutil.NopCloser(bytes.NewReader(data))
```

Reply:
see my commit reference in the PR if you want to try

Reply:
Thanks :fireworks: